migrate_sqlite_to_postgresql.py

```diff
@@ -0,0 +1,357 @@
+                    elif 'INTEGER' in sqlite_type:
+                        col_def += 'INTEGER'
+                    elif 'VARCHAR' in sqlite_type or 'TEXT' in sqlite_type or 'STRING' in sqlite_type:
+                        col_def += 'TEXT'
+                    elif 'DATETIME' in sqlite_type:
+                        col_def += 'TIMESTAMP'
+                    elif 'JSON' in sqlite_type:
+                        col_def += 'JSONB'
+                    else:
+                        col_def += 'TEXT'
+
+                    if not col.get('nullable', True):
+                        col_def += ' NOT NULL'
+
+                    col_definitions.append(col_def)
+
+                # Add primary key constraint (only if not already handled by SERIAL)
+                if primary_keys and primary_keys.get('constrained_columns'):
+                    pk_cols = primary_keys['constrained_columns']
+                    # Only add explicit PRIMARY KEY constraint if we have multiple columns
+                    # or if the single PK column is not SERIAL (SERIAL implies PRIMARY KEY)
+                    if len(pk_cols) > 1:
+                        pk_col_list = ', '.join([f'"{col}"' for col in pk_cols])
+                        col_definitions.append(f'PRIMARY KEY ({pk_col_list})')
+                    elif len(pk_cols) == 1:
+                        # Check if the single PK column is SERIAL
+                        pk_col_name = pk_cols[0]
+                        pk_col_info = next((col for col in columns if col['name'] == pk_col_name), None)
+                        if pk_col_info:
+                            sqlite_type = str(pk_col_info['type']).upper()
+                            if not ('INTEGER' in sqlite_type):
+                                # Not INTEGER, so not SERIAL, need explicit PRIMARY KEY
+                                col_definitions.append(f'PRIMARY KEY ("{pk_col_name}")')
+
+                # Add unique constraints
+                for constraint in unique_constraints:
+                    if constraint.get('column_names'):
+                        unique_cols = ', '.join([f'"{col}"' for col in constraint['column_names']])
+                        col_definitions.append(f'UNIQUE ({unique_cols})')
+
+                create_table_sql = f'CREATE TABLE "{table_name}" (\n  ' + ',\n  '.join(col_definitions) + '\n)'
+                pg_conn.execute(text(create_table_sql))
+
+                # Create indexes
+                for index in indexes:
+                    if not index.get('unique', False):
+                        index_cols = ', '.join([f'"{col}"' for col in index['column_names']])
+                        index_name = f'ix_{table_name}_{"_".join(index["column_names"])}'
+                        create_index_sql = f'CREATE INDEX "{index_name}" ON "{table_name}" ({index_cols})'
+                        pg_conn.execute(text(create_index_sql))
+
+            pg_conn.commit()
+
+        logger.info("✅ Schema copied successfully")
```

Comment:
Couldn't the schema be initialized by Alembic with a simple `alembic upgrade head`? The above code looks extremely fragile.

Reply:
I tried that initially and it kept failing due to mismatches with the alembic_version stamps in the database. This turned out to be the way that got the data copied and didn't cause a failed migration on server start up.